backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/dsl/impl/LuceneByteVectorFieldTypeOptionsStep.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.types.dsl.impl;
+
+import java.util.Arrays;
+
+import org.hibernate.search.backend.lucene.types.codec.impl.Storage;
+import org.hibernate.search.engine.backend.types.VectorSimilarity;
+
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.document.KnnByteVectorField;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.index.VectorEncoding;
+
+class LuceneByteVectorFieldTypeOptionsStep
+		extends AbstractLuceneVectorFieldTypeOptionsStep<LuceneByteVectorFieldTypeOptionsStep, byte[]> {
+
+	LuceneByteVectorFieldTypeOptionsStep(LuceneIndexFieldTypeBuildContext buildContext) {
+		super( buildContext, byte[].class );
+	}
+
+	@Override
+	protected LuceneByteVectorFieldTypeOptionsStep thisAsS() {
+		return this;
+	}
+
+	@Override
+	protected AbstractVectorFieldCodec<byte[], ?> createCodec(VectorSimilarity vectorSimilarity, Integer dimension,
+			Storage storage, byte[] indexNullAsValue, KnnVectorsFormat knnVectorsFormat) {
+		return new ByteVectorCodec( vectorSimilarity, dimension, storage, indexNullAsValue, knnVectorsFormat );
+	}
+
+	private static class ByteVectorCodec extends AbstractVectorFieldCodec<byte[], byte[]> {
+		public ByteVectorCodec(VectorSimilarity vectorSimilarity, Integer dimension, Storage storage, byte[] indexNullAsValue,
+				KnnVectorsFormat knnVectorsFormat) {
+			super( vectorSimilarity, dimension, storage, indexNullAsValue, knnVectorsFormat );
+		}
+
+		@Override
+		public byte[] decode(IndexableField field) {
+			KnnByteVectorField byteVectorField = (KnnByteVectorField) field;
+			return byteVectorField.vectorValue();
```

Comment:
Does this even work? I would be surprised, since the `toStoredField` method does not return a `KnnByteVectorField`.